File: client/src/main/java/com/sio/pizzeria/DTO/HistoriqueProduitDTO.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.sio.pizzeria.DTO;

/**
 *
 * @author Lenovo P50
 */
public class HistoriqueProduitDTO {
     

    private int idHistorique;
    private String nomProduit;
    private float prixProduit;
    private int idType;

    public HistoriqueProduitDTO(String nomProduit, float prixProduit, int idType) {
        this.nomProduit = nomProduit;
        this.prixProduit = prixProduit;
        this.idType = idType;
    }

    
    
    public int getIdHistorique() {
        return idHistorique;
    }

    public void setIdHistorique(int idHistorique) {
        this.idHistorique = idHistorique;
    }

    public String getNomProduit() {
        return nomProduit;
    }

    public void setNomProduit(String nomProduit) {
        this.nomProduit = nomProduit;
    }

    public float getPrixProduit() {
        return prixProduit;
    }

    public void setPrixProduit(int prixProduit) {
        this.prixProduit = prixProduit;
    }

    public int getIdType() {
        return idType;
    }

    public void setIdType(int idType) {
        this.idType = idType;
    }

    
}

File: server/DTO/HistoriqueProduitDTO.php
<?php


class HistoriqueProduitDTO implements JsonSerializable
{
    private $idHistorique;
    private $nomProduit;
    private $prixProduit;
    private $idType;


    public function __construct($nomProduit, $prixProduit, $idType)
    {
        $this->nomProduit = $nomProduit;
        $this->prixProduit = $prixProduit;
        $this->idType = $idType;
    }

    public function getIdHistorique()
    {
        return $this->idHistorique;
    }

    public function setIdHistorique($idHistorique)
    {
        $this->idHistorique = $idHistorique;
    }

    public function getNomProduit()
    {
        return $this->nomProduit;
    }

    public function setNomProduit($nomProduit)
    {
        $this->nomProduit = $nomProduit;
    }

    public function getPrixProduit()
    {
        return $this->prixProduit;
    }

    public function setPrixProduit($prixProduit)
    {
        $this->prixProduit = $prixProduit;
    }
    public function getIdType()
    {
        return $this->idType;
    }

    public function setIdType($idType)
    {
        $this->idType = $idType;
    }

    public function jsonSerialize()
    {
        return array(
            'idHistorique' => $this->idHistorique,
            'nomProduit' => $this->nomProduit,
            'prixProduit' => $this->prixProduit,
            'idType' => $this->idType
        );
    }
}